src/libraries/Microsoft.Extensions.FileSystemGlobbing/tests/OrderedPatternMatchingTests.cs

```diff
@@ -0,0 +1,180 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using Microsoft.Extensions.FileSystemGlobbing.Tests.TestUtility;
+using Xunit;
+
+namespace Microsoft.Extensions.FileSystemGlobbing.Tests
+{
+    public class OrderedPatternMatchingTests : PatternMatchingTests
+    {
+        protected override bool PreserveFilterOrder => true;
+
+        [Fact]
+        public void ComplexPatternSequence_SimpleFilters()
+        {
+            var scenario = new FileSystemGlobbingTestContext(@"c:/project/", true)
+                .Include("A/*")
+                .Exclude("A/B/*")
+                .Include("C/*")
+                .Files(
+                    "A/Program.cs",
+                    "A/B/foo.txt",
+                    "C/README.md"
+                )
+                .Execute();
+
+            scenario.AssertExact(
+                "A/Program.cs",
+                "C/README.md"
+            );
+        }
+
+        [Fact]
+        public void ComplexPatternSequence_DeeplyNestedFilters()
+        {
+            var scenario = new FileSystemGlobbingTestContext(@"c:/project/", true)
+                .Include("**/*.cs")                                   // Include all .cs files
+                .Exclude("**/obj/**/*")                               // Exclude all in obj/
+                .Include("lib/generated/**/*.cs")                     // Re-include generated code
+                .Exclude("**/*Tests.cs")                              // Exclude unit tests
+                .Include("tests/manual/*Tests.cs")                    // Re-include manual test files
+                .Exclude("legacy/**/*")                               // Exclude legacy code
+                .Files(
+                    "Program.cs",
+                    "obj/Temp.cs",
+                    "lib/generated/AutoGen.cs",
+                    "Tests/Unit/MathTests.cs",
+                    "tests/manual/VisualTests.cs",
+                    "legacy/OldStuff.cs",
+                    "README.md"
+                )
+                .Execute();
+
+            scenario.AssertExact(
+                "Program.cs",
+                "lib/generated/AutoGen.cs",
+                "tests/manual/VisualTests.cs"
+            );
+        }
+
+        [Fact]
+        public void MixOfPatternsAcrossVariousFileTypes()
```

Comment:
Seems like Complex means `Include -> Exclude -> Include` or `Exclude -> Include -> Exclude`, if so, this test should also be prefixed as Complex?

Reply:
Removed the `Complex` naming.